modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java

```diff
@@ -562,6 +562,18 @@ private void _computeFolders() {
 					}
 				}
 			}
+
+			if (_folder != null) {
+				List<Long> ancestors = _folder.getAncestorFolderIds();
+
+				if ((_folderId != _rootFolderId) &&
+					(_rootFolderId !=
+						DLFolderConstants.DEFAULT_PARENT_FOLDER_ID) &&
+					!ancestors.contains(_rootFolderId)) {
+
+					throw new NoSuchFolderException();
+				}
+			}
```

Comment:
We can save computation by restructuring these nested IFs like this:
```java
			if ((_folder != null) && (_folderId != _rootFolderId) &&
				(_rootFolderId != DLFolderConstants.DEFAULT_PARENT_FOLDER_ID)) {
				List<Long> ancestors = _folder.getAncestorFolderIds();

				if (!ancestors.contains(_rootFolderId)) {
					throw new NoSuchFolderException();
				}
			}
```

Reply:
Thank you @marcogalluzzi, I will make that change.